src/bindings/python/src/pyopenvino/core/common.cpp

```diff
@@ -118,35 +118,46 @@ const TensorIndexMap cast_to_tensor_index_map(const py::dict& inputs) {
 
 namespace string_helpers {
 
+namespace {
+auto find_last_not_null(const char* str, size_t length) {
+    return std::find_if(std::make_reverse_iterator(str + length),
+                        std::make_reverse_iterator(str),
+                        [](const auto& c) {
+                            return c != '\0';
+                        })
+        .base();
+}
+}  // namespace
+
 py::array bytes_array_from_tensor(ov::Tensor&& t) {
     if (t.get_element_type() != ov::element::string) {
         OPENVINO_THROW("Tensor's type must be a string!");
     }
     auto data = t.data<std::string>();
+
     auto max_element = std::max_element(data, data + t.get_size(), [](const std::string& x, const std::string& y) {
         return x.length() < y.length();
     });
     auto max_stride = max_element->length();
     auto dtype = py::dtype("|S" + std::to_string(max_stride));
     // Adjusting strides to follow the numpy convention:
     py::array array;
-    auto new_strides = t.get_strides();
-    if (new_strides.size() == 0) {
-        array = py::array(dtype, t.get_shape(), {});
+
+    if (auto new_strides = t.get_strides(); new_strides.empty()) {
+        array = py::array(dtype, t.get_shape(), std::move(new_strides));
     } else {
         auto element_stride = new_strides[new_strides.size() - 1];
         for (size_t i = 0; i < new_strides.size(); ++i) {
             new_strides[i] = (new_strides[i] / element_stride) * max_stride;
         }
-        array = py::array(dtype, t.get_shape(), new_strides);
+        array = py::array(dtype, t.get_shape(), std::move(new_strides));
```

Comment:
May be we can even simplify this if-else?
like 
```cpp
auto new_strides = t.get_strides();
if (!new_strides.empty()) {
        auto element_stride = new_strides[new_strides.size() - 1];
        for (size_t i = 0; i < new_strides.size(); ++i) {
            new_strides[i] = (new_strides[i] / element_stride) * max_stride;
}
array = py::array(dtype, t.get_shape(), std::move(new_strides));
```

Reply:
The current solution has benefit that `new_strides` has  lifetime for `if` only. Is always better to limit scope of objects. But refactored to avoid using move at all.